tests/integration/incremental_unique_id_test/test_incremental_unique_id.py

```diff
@@ -0,0 +1,270 @@
+            self.unique_schema(),
+            seed
+        )
+
+        self.run_sql_file(Path('seeds') / Path(update_sql_file + '.sql'))
+        return len(self.run_sql(row_count_query, fetch='all'))
+
+    def update_incremental_model(self, incremental_model):
+        '''update incremental model after the seed table has been updated'''
+        model_result_set = self.run_dbt(['run', '--select', incremental_model])
+        return len(model_result_set)
+
+    def setup_test(self, seed, incremental_model, update_sql_file):
+        '''build a test case and return values for assertions'''
+        (seed_count, model_count) = self.count_test_relations(
+            seed=seed, incremental_model=incremental_model
+        )
+        seed_rows = self.get_seed_rows_after_update(
+            seed=seed, update_sql_file=update_sql_file
+        )
+        inc_test_model_count = self.update_incremental_model(
+            incremental_model=incremental_model
+        )
+
+        return (seed_count, model_count, seed_rows, inc_test_model_count)
+
+    def run_test(self, expected_fields, test_case_fields):
+        '''Invoke assertions to verify correct build functionality'''
+        # 1. test seed(s) should build afresh
+        self.assertEqual(
+            expected_fields.seed_count, test_case_fields.seed_count
+        )
+        # 2. test model(s) should build afresh
+        self.assertEqual(
+            expected_fields.model_count, test_case_fields.model_count
+        )
+        # 3. seeds should have intended row counts post update
+        self.assertEqual(
+            expected_fields.seed_rows, test_case_fields.seed_rows
+        )
+        # 4. incremental test model(s) should be updated
+        self.assertEqual(
+            expected_fields.inc_test_model_count,
+            test_case_fields.inc_test_model_count
+        )
+        # 5. extra incremental model(s) should be built; optional since
+        #   comparison may be between an incremental model and seed
+        if (expected_fields.opt_model_count and
+            test_case_fields.opt_model_count):
+            self.assertEqual(
+                expected_fields.opt_model_count,
+                test_case_fields.opt_model_count
+            )
+        # 6. result table should match intended result set (itself a relation)
+        self.assertTablesEqual(
+            expected_fields.relation, test_case_fields.relation
+        )
+
+    def stub_expected_fields(
+        self, relation, seed_rows, opt_model_count=None
+    ):
+        return TestResults(
+            seed_count=1, model_count=1, seed_rows=seed_rows,
+            inc_test_model_count=1, opt_model_count=opt_model_count,
+            relation=relation
+        )
+
+class TestIncrementalWithoutUniqueKey(TestIncrementalUniqueKey):
+    @use_profile('snowflake')
+    def test__snowflake_no_unique_keys(self):
+        '''with no unique keys, seed and model should match'''
+        seed='seed'
+        seed_rows=8
```

Comment:
I saw that most seed rows are either 7 or 8, do you think it would make sense to modify the test case so that we have more diverse number for different situation?

Reply:
I could. It would just mean adding another seed file. Not sure if that's needed now. Really, I set this up now so that we could easily add things in the future. See, this incremental test is mostly a matter of the combination between unique keys and either a simple insert/a merge. And so, my thinking went, "We only need two test scenarios; the model configs are where a lot of the testing will take place."

That make your concerns go away or nah, you still think we should diversify out the underlying test scenarios?

Reply:
I think this is good enough!